modules/miner/update.go

```diff
@@ -1,11 +1,205 @@
 package miner
 
 import (
-	"github.com/NebulousLabs/Sia/encoding"
 	"github.com/NebulousLabs/Sia/modules"
 	"github.com/NebulousLabs/Sia/types"
 )
 
+func (m *Miner) getNewSplitSets(diff *modules.TransactionPoolDiff) []*mapElement {
+	// Split the new sets and add the splits to the list of transactions we pull
+	// form.
+	newElements := make([]*mapElement, 0)
+	for _, newSet := range diff.AppliedTransactions {
+		// Split the sets into smaller sets, and add them to the list of
+		// transactions the miner can draw from.
+		//
+		// TODO: Split the one set into a bunch of smaller sets using the cp4p
+		// splitter.
+		m.setCounter++
+		m.fullSets[newSet.ID] = []int{m.setCounter}
+		var size uint64
+		var totalFees types.Currency
+		for i := range newSet.IDs {
+			size += newSet.Sizes[i]
+			for _, fee := range newSet.Transactions[i].MinerFees {
+				totalFees = totalFees.Add(fee)
+			}
+		}
+		// We will check to see if this splitSet belongs in the block.
+		s := &splitSet{
+			size:         size,
+			averageFee:   totalFees.Div64(size),
+			transactions: newSet.Transactions,
+		}
+
+		elem := &mapElement{
+			set:   s,
+			id:    splitSetID(m.setCounter),
+			index: 0,
+		}
+
+		m.splitSets[splitSetID(m.setCounter)] = s
+		newElements = append(newElements, elem)
+	}
+	return newElements
+}
+
+// addNewTxns adds new unconfirmed transactions to the miner's transaction
+// selection.
+func (m *Miner) addNewTxns(diff *modules.TransactionPoolDiff) {
+	// Get new splitSets (in form of mapElement)
+	newElements := m.getNewSplitSets(diff)
+
+	// Place each elem in one of the MapHeaps
+	for _, elem := range newElements {
```

Comment:
more efficient to use the construction `for i := 0; i < len(newElements); i++ {` - does less memory copying.